components/Modals/BuyingGuideModal.tsx

```diff
@@ -51,7 +52,22 @@ export default function BuyingGuide({
     }
   }, [isCopiedTooltipOpen]);
 
-  const USDC = formatter.format(Number(balance?.formatted) || 0);
+  useEffect(() => {
+    const formatted = Number(balance?.formatted);
+    const val = BigNumber.from(balance?.value);
+    const currBuyAmt = utils
+      .parseUnits(buyAmount.toString(), 6)
```

Comment:
as it's copy paste - let's extract it to method :P

Reply:
actually we don't need the short circuit since it's moved into the mount check now, I can just remove the other one